lib/internal/Magento/Framework/App/Cache/FlushCacheByTags.php

```diff
@@ -95,18 +85,18 @@ public function aroundDelete(AbstractResource $subject, \Closure $proceed, Abstr
      * Clean cache by tags
      *
      * @param string[] $tags
-     * @return void
      */
     private function cleanCacheByTags(array $tags): void
     {
         if (!$tags) {
             return;
         }
+        $uniqueTags = null;
         foreach ($this->cacheList as $cacheType) {
             if ($this->cacheState->isEnabled($cacheType)) {
                 $this->cachePool->get($cacheType)->clean(
                     \Zend_Cache::CLEANING_MODE_MATCHING_ANY_TAG,
-                    \array_unique($tags)
+                    $uniqueTags = $uniqueTags ?? \array_unique($tags)
```

Comment:
Why? Is this really needed?

Reply:
Previously the code was performing array_unique in a loop (slow!) Caching the value locally makes sense.

Reply:
I agree, so let's do it once before and pass the variable to the for each. Write easy to read code and easy to debug code. Debugging that will be a pain in the future.